scs2-simulation/src/main/java/us/ihmc/scs2/simulation/physicsEngine/contactPointBased/ContactPointBasedRobotPhysics.java

```diff
@@ -39,6 +45,17 @@ public class ContactPointBasedRobotPhysics
    private final ForwardDynamicsCalculator forwardDynamicsCalculator;
    private RobotJointWrenchCalculator jointWrenchCalculator;
 
+   private final JointMatrixIndexProvider indexProvider;
+
+   /** The joint torques from the robot's controller manager, corresponding to joint torques commanded by any controllers. */
+   private final YoMatrix jointsTauController;
+
+   /** The joint torques imposed by physics simulation, consisting of damping + soft enforcement of joint limits. */
+   private final YoMatrix jointsTauSimulationEffects;
+
+   /** The resultant joint torques to be used in the forward dynamics calculation (i.e. the calculation for the result of the simulation). */
+   private final YoMatrix jointsTau;
+
```

Comment:
☝🏻 `YoMatrix`s for tracking torque contributions from the controller, and what I'm terming simulation effects: damping and limit enforcement. Just before calling forward dynamics, we sum these and pass them as input to FD.